src/pyrokinetics/local_geometry/local_geometry.py

```diff
@@ -1,35 +1,138 @@
+from typing_extensions import Self, TypeAlias
 
 from ..constants import pi
 from ..decorators import not_implemented
 from ..equilibrium import Equilibrium
 from ..factory import Factory
 from ..typing import ArrayLike
+from ..units import PyroQuantity as Quantity
 from ..units import ureg as units
 
 if TYPE_CHECKING:
     import matplotlib.pyplot as plt
 
     from ..normalisation import SimulationNormalisation as Normalisation
 
+Float: TypeAlias = Union[float, Quantity]
+Array: TypeAlias = Union[NDArray[np.float64], Quantity]
+
+
+class LocalGeometry:
+
+    psi_n: Float
+    r"""Normalised :math:`\Psi`"""
+
+    rho: Float
+    r"""Normalised minor radius.
+
+    :math:`r/a`, where :math:`a` is the minor radius of the last closed flux
+    surface.
+    """
+
+    a_minor: Float
+    """Minor radius of the last closed flux surface."""
+
+    Rmaj: Float
+    r"""Normalised major radius.
+
+    :math:`R/a`, where :math:`a` is the minor radius of the last closed flux
+    surface.
+    """
+
+    Z0: Float
+    r"""Normalised vertical position of midpoint.
+
+    :math:`Z_{mid}/a`, where :math:`a` is the minor radius of the last closed
+    flux surface.
+    """
+
+    Fpsi: Float
+    """Torodial field function"""
+
+    B0: Float
+    r"""Toroidal field at major radius.
+
+    :math:`F_\psi/R`
+    """
+
+    bunit_over_b0: Float
+    r"""Ratio of GACODE normalising field to :math:`B_0`.
+
+    :math:`B_{unit}=q/r \partial \psi/\partial r`
+    """
+
+    dpsidr: Float
+    r""":math:`\partial \psi / \partial r`"""
+
+    q: Float
+    """Safety factor"""
 
-def default_inputs():
-    # Return default args to build a LocalGeometry
-    # Uses a function call to avoid the user modifying these values
-    return {
+    shat: Float
+    r"""Magnetic shear :math:`r/q \partial q/ \partial r`"""
+
+    beta_prime: Float
+    r""":math:`\beta = 2 \mu_0 \partial p \partial \rho 1/B_0^2`"""
+
+    R_eq: Array
+    """Equilibrium ``R`` data used for fitting"""
+
+    Z_eq: Array
+    """Equilibrium ``Z`` data used for fitting"""
+
+    b_poloidal_eq: Array
+    """Equilibrium ``B_poloidal`` data used for fitting"""
+
+    theta_eq: Float
+    """Theta values for equilibrium data"""
+
+    R: Array
+    """Fitted ``R`` data"""
+
+    Z: Array
+    """Fitted ``Z`` data"""
+
+    b_poloidal: Array
+    """Fitted ``B_poloidal`` data"""
+
+    theta: Float
+    """Fitted theta data"""
+
+    dRdtheta: Array
+    r"""Derivative of fitted :math:`R` w.r.t :math:`\theta`"""
+
+    dRdr: Array
+    r"""Derivative of fitted :math:`R` w.r.t :math:`r`"""
+
+    dZdtheta: Array
+    r"""Derivative of fitted :math:`Z` w.r.t :math:`\theta`"""
+
+    dZdr: Array
+    r"""Derivative of fitted :math:`Z` w.r.t :math:`r`"""
+
+    DEFAULT_INPUTS: ClassVar[Dict[str, float]] = {
         "psi_n": 0.5,
         "rho": 0.5,
         "Rmaj": 3.0,
         "Z0": 0.0,
         "a_minor": 1.0,
         "Fpsi": 0.0,
-        "B0": None,
+        "B0": 0.0,
```

Comment:
I'm nervous about setting this to 0 as we sometimes to `1/B0`...

Reply:
In some ways it would make more sense to _not_ have a default value at all for basically all of these

Reply:
@ZedThree I'd like to eventually get these classes to a point where none of the inputs are optional (besides things like `delta` in `LocalGeomtryMXH`), but the option to build one with sensible defaults could still be achieved with a class method, e.g. `LocalGeometry::default()`, as that would make testing a lot easier.

Reply:
I think we usually have checks for `B0` usually like 

```python
if local_geometry.B0:
    beta = 1 / local_geometry.B0**2
else:
    beta = 0.0
```

Hence the default of `None` as having `beta=0.0` by default is safest.